### PR TITLE
fix(worker): clear `ngx.ctx` before running handler

### PR DESCRIPTION
**Pull request by ADD-SP**
# Please REBASE AND MERGE

# Summary

Clear `ngx.ctx` before running the handler to avoid race conditions.

[KAG-3215]

[KAG-3215]: https://konghq.atlassian.net/browse/KAG-3215?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/resty/timerng/thread/worker.lua [hunk 1]
```diff
@@ -165,6 +168,8 @@ local function thread_body(context, self,
             goto continue
         end
 
+        utils_table_clear(ngx.ctx)
```
Comment (tzssangglass) [inline lib/resty/timerng/thread/worker.lua:171]: Will this affect the current request? It seems like a resty library is manipulating the current request ctx.

Comment (ADD-SP) [reply lib/resty/timerng/thread/worker.lua:171]: This line always runs in the `timer` context, I think the request context should not be accessed in a timer.

Comment (tzssangglass) [reply lib/resty/timerng/thread/worker.lua:171]: clear, the timer has its own request ctx.

**Comment by chronolaw** [comment]
I saw some luacheck failings, could we fix them to make linter happy?

File: spec/07-bugs_spec.lua [hunk 2]
```diff
@@ -89,4 +89,66 @@ insulate("bugs of every timer | ", function ()
...
+            return true
+        end)
+
+    end)
+
+    before_each(function ()
+        update_time()
+    end)
+
+    it("", function ()
```
Comment (chronolaw) [inline spec/07-bugs_spec.lua:123]: Why does it have no description? is it OK?

Comment (ADD-SP) [reply spec/07-bugs_spec.lua:123]: It is intentional, the description in the `describe ()` is enough.

**Comment by ADD-SP** [comment]
> I saw some luacheck failings, could we fix them to make linter happy?

Fixed by bump the version of linter.

**Review by chronolaw — APPROVED** [review]

**Review by chobits — APPROVED** [review]

**Pull request merged**